hbase-server/src/main/java/org/apache/hadoop/hbase/io/FileLink.java

```diff
@@ -405,14 +424,31 @@ public Path getAvailablePath(FileSystem fs) throws IOException {
    * @throws IOException on unexpected error.
    */
   public FileStatus getFileStatus(FileSystem fs) throws IOException {
+    IOException exception = null;
     for (int i = 0; i < locations.length; ++i) {
       try {
         return fs.getFileStatus(locations[i]);
-      } catch (FileNotFoundException e) {
-        // Try another file location
+      } catch (FileNotFoundException | AccessControlException e) {
+        exception = handleGetFileStatusException(e, exception);
```

Comment:
Same as the above.  Why not share the above handleTryOpenException ?

Reply:
The handleTryOpenException is a method of inner class, and the tow handle exceptions method is not the same?